File: nim/raindrops/raindrops.nim
proc add(n: int, factor: int, word: string): string =
  if n mod factor == 0:
    word
  else:
    ""

proc convert*(n: int): string =
  result = add(n, 3, "Pling") & 
    add(n, 5, "Plang") & 
    add(n, 7, "Plong")

  if result == "":
    result = $(n)

File: ocaml/raindrops/raindrops.ml
let word_for factor word n =
	if n mod factor == 0 then
		word
	else
		""

let raindrop i =
	let
		phrase =
			[
				word_for 3 "Pling" i;
				word_for 5 "Plang" i;
				word_for 7 "Plong" i;
			]
			|> String.concat ""
	in
	if phrase = "" then
		string_of_int i
	else
		phrase


